packages/server/graphql/mutations/helpers/handleCompletedStage.ts

```diff
@@ -16,7 +24,47 @@ const handleCompletedRetrospectiveStage = async (
   dataLoader: DataLoaderWorker
 ) => {
   if (stage.phaseType === REFLECT || stage.phaseType === GROUP) {
-    const data = await removeEmptyReflections(meeting)
+    const data: Record<string, any> = await removeEmptyReflections(meeting)
+
+    if (stage.phaseType === REFLECT) {
+      const r = await getRethink()
+      const now = new Date()
+      const reflectionGroups = await dataLoader
+        .get('retroReflectionGroupsByMeetingId')
+        .load(meeting.id)
+      const reflections = await r
+        .table('RetroReflection')
+        .getAll(meeting.id, {index: 'meetingId'})
+        .filter({isActive: true})
+        .orderBy('createdAt')
+        .run()
+      const {reflectionGroupMapping, autoGroupThreshold, nextThresh} = groupReflections(
+        reflections,
+        AUTO_GROUPING_THRESHOLD
+      )
+      console.log(`autoGroupThreshold = ${autoGroupThreshold}; nextThresh = ${nextThresh}`)
+
+      reflectionGroups.sort((groupA, groupB) => {
```

Comment:
slice first! here you're mutating an array provided by a dataloader. that means ANY other function that is using that same dataloader is going to get mutated & depending on when this runs, that means it'll be nondeterministic